ado/ipacheckenum.ado

```diff
@@ -534,10 +533,23 @@ program define ipacheckenumvarstats
 		isid `enum'
 
 		// Convert enum variable to string
-		rename `enum' `enum'_temp
-		decode `enum'_temp, gen(`enum')
-		drop `enum'_temp
-		
+		cap confirm numeric variable `enum' 
+		// numeric variables
+		if _rc==0  {
+			local label_name :value label `enum'    
+			// with labels
+			if ("`label_name'" != ""){
+			di in r "label"
+				rename `enum' `enum'_temp
+				decode `enum'_temp, gen(`enum')
+				drop `enum'_temp
+			}
+			else if ("`label_name'" == "") {
+				di in r "no label"

```

Comment:
@isabelonate would it be OK to take out these references to display "no label" or "label" in red? I assume you were using these to test the execution of the if conditions... unless you intended for them to serve another purpose?